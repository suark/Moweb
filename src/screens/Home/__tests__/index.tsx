import React from 'react';
import renderer from 'react-test-renderer';
import { Home } from '..';
import { mockNavigation } from '../../../mocks/Setup';

it('should render correctly', () => {
  const component = renderer.create(<Home />);
  expect(component.toJSON()).toMatchSnapshot();
});

it('navigates to settings', () => {
  const component = renderer.create(<Home />);
  const button = component.root.findByProps({ testID: 'settings' });
  button.instance.props.onPress();
  expect(mockNavigation.navigate).toBeCalledWith('Settings');
});
