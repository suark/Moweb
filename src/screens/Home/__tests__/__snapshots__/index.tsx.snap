// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <View
    accessibilityRole="button"
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
    testID="settings"
  >
    <View
      style={
        Array [
          Object {},
        ]
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#007AFF",
              "fontSize": 18,
              "margin": 8,
              "textAlign": "center",
            },
            Object {
              "color": "rgb(0, 122, 255)",
            },
          ]
        }
      >
        Go to Settings
      </Text>
    </View>
  </View>
</View>
`;
